pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -54,12 +60,44 @@ func updateConditions(msConditions []v1alpha1.Condition, prom monv1.Prometheus,
 				reconciled.LastTransitionTime = metav1.Now()
 			}
 			updatedConditions = append(updatedConditions, reconciled)
+		case v1alpha1.ResourceDiscoveryCondition:
```

Comment:
doesn't this mean, that a `ms` that is already deployed prior to this feature  will  never  have this condition set? Instead shouldn't we set all 3 conditions irrespective of them them having the condition?

Reply:
Hm yeah good point.